lib/order.rb

```diff
@@ -0,0 +1,40 @@
+require_relative 'send_sms'
+require_relative 'dishes'
+
+class Order
+
+  attr_reader :selection, :dishes, :sms
+
+  def initialize(dishes = Dishes.new, sms = SendSMS.new)
+    @dishes = dishes
+    @sms = sms
+  end
+
+  def view_dishes
+    dishes.view
+  end
+
+  def create(selection)
+    @selection = selection
+  end
+
+  def verify(user_total)
+    error_msg = "Order Error - Total does not match to price list"
+    raise error_msg if calculate_total != user_total
+    sms.send
+    "Order placed and sms message sent to confirm delivery"
+  end
+
+  private
+
+  def calculate_total
+    total = 0
+    dish_split = selection.split(",")
+    dish_split.each do |item|
+      dish = item.split("*")[0]
+      quantity = item.split("*")[1].to_i
+      total += dishes.price_list[dish] * quantity
+    end
+    total
+  end
+end
```

Comment:
Easy to follow methods, named well. It's clear what the code is doing. 
It's nice that the user can add multiple items at once. I did not think of that!
Maybe this could just be formatted slightly differently so it's less specific?  
Would be cool if the user could see the calculated sub-total of their order before verifying but it's handy that they have to type the total themselves to ensure it's checked.

Reply:
Maybe there should be a separate restaurant/takeaway class which can create a new instance of both order and dishes/menu?